src/conn.rs

```diff
@@ -0,0 +1,59 @@
+#[cfg(all(not(doc), feature = "native-tls", feature = "rustls-tls"))]
+compile_error!(r#"Only one of "native-tls" and "rustls-tls" may be enabled for TLS support"#);
+#[cfg(feature = "native-tls")]
+use native_tls::TlsStream;
+#[cfg(all(not(doc), feature = "rustls-tls"))]
+use rustls_connector::TlsStream;
+use std::io::{Read, Write};
+use std::net::TcpStream;
+
+/// A generic stream wrapper around either a Tcp or a Tls connection
+#[derive(Debug)]
+pub enum ConnType {
```

Comment:
Couldn't we just replace this whole thing with
```rust
pub trait ImapConnection: Read + Write {}
impl<T> ImapConnection for T where T: Read + Write {}

pub type ConnType = Box<dyn ImapConnection>;
```
I think that'd be much nicer, and it should also allow the two TLS implementations to co-exist rather than being mutually exclusive.

Reply:
Furthermore, by doing this, we should be able to make this just be part of the standard builder.